python-backend/app/api/parties/party_appt/models/mine_party_appt.py

```diff
@@ -98,6 +103,36 @@ def find_parties_by_mine_party_appt_type_code(cls, code):
         except ValueError:
             return None
 
+    @classmethod
+    def find_manager_history_by_mine_no(cls, mine_no):
+        # send internal API call to fetch matching mine record
+        mines_url = current_app.config['MINES_URL'] + '/mines/' + str(mine_no)
+        auth_headers = request.headers.get('Authorization')
+        if not auth_headers:
+            return None, 401, 'Unauthorized'
+
+        headers = {'Authorization': auth_headers}
+        response = requests.get(url=mines_url, headers=headers)
+        if not response:
+            return None, 400, 'Unexpected error'
```

Comment:
This would be an internal web call 400., to the one sent to the server was fine. I feel like this should return a 500.

Reply:
That's fair. The two places I put a 400 marked errors that I wasn't totally sure about how to describe. Open to input on either of those and will update the codes accordingly